### PR TITLE
Compiler's -S output and `-save-temps=obj` loses MVE feature which le…

### PR DESCRIPTION
**Pull request by simpal01**
At the moment clang is not generating the correct FPU directive for Arm V8.1m Mainline CPU's when assembled by -S. So when compiled with -S output and `-save-temps=obj` loses MVE feature which leads to assembler error.

There is a mismatch in the default FPU option that Arm V8.1-m CPUs define and what clang generates. For example, Clang generates fpv5-d16 for M85 and what is defined in ARMTargetParser.def is FK_FP_ARMV8_FULLFP16_D16.

This patch add cases for handling this FPU directive and thus makes clang generates the right FPU directives for Arm V8.1-m CPUs.

### CODE REVIEW

**Review by smithp35 — APPROVED** [review]
LGTM, although I'm a bit close to the implementation. Would be good if we can some more eyes on it.

**Comment by amilendra** [comment]
I'd rephrase this part of the commit message slightly.
> There is a change in the default FPU option that Arm V8.1-m CPUs defines and what the clang generates. For example, Clang generates fpv5-d16 for M85 and what defined in ARMTargetParser.def is FK_FP_ARMV8_FULLFP16_D16.

Please check if the below is what you meant.

There is a mismatch in the default FPU option that Arm V8.1-m CPUs define and what clang generates.
For example, Clang generates fpv5-d16 for M85 and what is defined in ARMTargetParser.def is FK_FP_ARMV8_FULLFP16_D16.

**Comment by amilendra** [comment]
The implementation LGTM. I haven't checked if the test covers the newly added control-flow path. But if Simi has checked it LGTM.

**Comment by simpal01** [comment]
> I'd rephrase this part of the commit message slightly.
> 
> > There is a change in the default FPU option that Arm V8.1-m CPUs defines and what the clang generates. For example, Clang generates fpv5-d16 for M85 and what defined in ARMTargetParser.def is FK_FP_ARMV8_FULLFP16_D16.
> 
> Please check if the below is what you meant.
> 
> There is a mismatch in the default FPU option that Arm V8.1-m CPUs define and what clang generates. For example, Clang generates fpv5-d16 for M85 and what is defined in ARMTargetParser.def is FK_FP_ARMV8_FULLFP16_D16.

Done

**Review by amilendra — APPROVED** [review]
